File: DEMO-SERVER.Core/DAL/Entities/Transaction/Order.cs
using DEMO.Core.DAL.Entities.Common;
using DEMO.Core.DAL.Entities.Master;
using DEMO_SERVER.Core.DAL.Entities.Transaction;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace DEMO.Core.DAL.Entities.Transaction
{
    public class Order : BaseEntity
    {
        public string OrderNo { get; set; }
        public long CustomerId { get; set; }
        public PaymentType PaymentMethod { get; set; }
        public double Total { get; set; }

        public List<OrderDetails> Details { get; set; }

        [ForeignKey("CustomerId")]
        public virtual Customer Customer { get; set; }

        public enum PaymentType
        {
            Cash = 0,
            Credit = 1
        }
    }
}

File: Demo-App.web/src/pages/transaction/order/model/order.ts
import { BaseEntity } from "src/core/model/BaseEntity";
import { Customer } from "src/pages/master-data/customer/model/customer";
import { Item } from "src/pages/master-data/item/model/item";

export interface Order extends BaseEntity{
    OrderNo: string;
    CustomerId: number;
    CustomerName: string;
    PaymentMethod: PaymentType;
    Total: number;

    Details: Array<OrderDetails>;
    Customer: Customer;
}

export interface OrderDetails extends BaseEntity{
    OrderId: number;
    ItemId: number;
    ItemName: string;
    Quantity: number;
    Price: number;

    Item: Item;
}

export enum PaymentType {
    Cash = 0,
    Credit = 1
}